mantidproject/mantid
# Add sequential and simultaneous-sequential fitting to the Muon Analysis 2 GUI

**Description of work.**
This PR re-introduces sequential fitting into the Muon Analysis 2 GUI, with the following additional functionality (those which are ticked have been completed):

- [x] Ability to run both sequential and sequential-simultaneous fits
- [x] A pre-populated table of workspaces used for the fits which can be re-ordered
- [x] Once a fit is complete the table is populated with values
- [x] A selected fit can be performed again if required
- [x] Fit parameters are adjustable in the table
- [x] User defined fit parameters are used as initial values for the fits
- [x] Carry over values from previous fit into the next one
- [x] Further unit tests





**To test:**
1.  Open the Muon Analysis 2 GUI
2.  Load in data, e.g MUSR 62260
3.  Go to fitting tab
4.  Add a fit function, select a workspace to do the fit over in the display workspace box
5.  Go to the sequential fitting tab
6. Ensure the following works
(a) Sequential fitting
(b) Selected single fits
7.  Switch the fitting to simultaneous in the fitting tab, and ensure the previous steps (a,b) work.


Fixes #27765. 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
